OPHD/UI/StringTable.cpp

```diff
@@ -0,0 +1,236 @@
+#include "StringTable.h"
+#include "../FontManager.h"
+#include "../Constants/UiConstants.h"
+#include <NAS2D/Utility.h>
+#include <stdexcept>
+
+const NAS2D::Color StringTable::Cell::ColorEmpty = NAS2D::Color::NoAlpha;
+
+StringTable::StringTable(std::size_t columns, std::size_t rows) : columns(columns), rows(rows)
+{
+	cells.resize(columns * rows);
+
+	defaultFont = NAS2D::Utility<FontManager>::get().font(constants::FONT_PRIMARY, constants::FONT_PRIMARY_NORMAL);
+	defaultTitleFont = NAS2D::Utility<FontManager>::get().font(constants::FONT_PRIMARY_BOLD, constants::FONT_PRIMARY_NORMAL);
+}
+
+void StringTable::draw(NAS2D::Renderer& renderer) const
+{
+	for (std::size_t i = 0; i < cells.size(); ++i)
+	{
+		const Cell& cell = cells.at(i);
+
+		NAS2D::Color textColor = cell.textColor != Cell::ColorEmpty ? cell.textColor : defaultTextColor;
+		NAS2D::Point<float> drawPosition = { position.x() + cell.textRelativePosition.x(), position.y() + cell.textRelativePosition.y() };
```

Comment:
If the relative position was changed to a `Vector`, this could be written as:
```cpp
const auto drawPosition = position + cell.textRelativePosition;
```